### DISCUSSION TITLE
Attachments default value

### DISCUSSION THREAD

**Discussion by peterhuppertz** [Budibase/budibase#8986] [Help]
I'm creating a form that can be used for both creating new and updating existing data. Since the combo datasource/repeater/update form does not work for this purpose (an empty repeater component hides all children) I tried to achieve this using a create form and setting the default values of all fields to the respective value from the datasource row.

This works fine with all form element types, except the attachment field, which does not have a default value property. I tried to use the "update field" action to set the value, without success. Does anyone know how to set the (default) value of an attachment field?

**Comment by melohagan** [comment]
Hey @peterhuppertz 

Attachment fields cannot have a default value. [Conditional UI](https://docs.budibase.com/docs/conditions) might work for you if you just need to show a placeholder image if no rows are present.

On you form something like: 
![Screenshot 2022-12-09 at 13 35 42](https://user-images.githubusercontent.com/101575380/206714213-e57ac49f-2a6a-4660-a51d-4be9a67784ed.png)

Then your placeholder image would have the same condition, but **Show component** instead of hide.

**Reply by peterhuppertz** [reply]
Thanks @melohagan. 
I actually ended up using conditional UI in the following way:
1) If there are existing attachments use a combination of Data Provider/Repeater/Attachment field to show them
<img width="173" alt="image" src="https://user-images.githubusercontent.com/5812004/207009842-8c093d8c-b2d9-4291-b4b5-7d59c4ef2abf.png">
2) If there are no existing attachments just show another attachment field
<img width="131" alt="image" src="https://user-images.githubusercontent.com/5812004/207009990-0ed0e14c-6ea3-40ab-a8f5-c751ce692944.png">

This solution works, however I feel it's adding unnecessary complexity. Would be great to have Repeaters showing empty form fields if there's no existing data.

**Reply by melohagan** [reply]
Thanks for getting back @peterhuppertz 

I've copied your setup and no conditional UI was needed...
![Screenshot 2022-12-12 at 09 36 49](https://user-images.githubusercontent.com/101575380/207011828-ee15b317-8c13-4073-b910-d444cee1dc1b.png)

When an attachment is present, it is shown. When there is no attachment for a row then it is empty. 

I may not be fully understanding what you are trying to achieve.

**Reply by peterhuppertz** [reply]
@melohagan When I use the Repeater element and I'm creating a new row there's no existing data from the Data Provider and I get "No rows found". How did you achieve having the Attachment field showing up if there's no existing data from the Data Provider?

<img width="344" alt="image" src="https://user-images.githubusercontent.com/5812004/207014024-817c2895-bef9-4be4-8aea-805010bda62a.png">

**Reply by melohagan** [reply]
Ah I see now.

I think you might like the new Side Panels that are coming out with the new release - hopefully today!
They combine *creating new and updating existing data* when you create the auto-generated screens.

![Screenshot 2022-12-12 at 10 46 19](https://user-images.githubusercontent.com/101575380/207026192-df5a67bf-e52e-46c7-a9cc-2e22ff703b51.png)

![Screenshot 2022-12-12 at 10 46 49](https://user-images.githubusercontent.com/101575380/207026291-6dd686d8-0802-456e-aff3-03fcd12b18f3.png)

![Screenshot 2022-12-12 at 10 47 02](https://user-images.githubusercontent.com/101575380/207026344-de2805f8-540f-4eab-851e-051fb098add3.png)
